src/main/java/javax/jmdns/impl/HostInfo.java

```diff
@@ -104,6 +106,11 @@ public static HostInfo newHostInfo(InetAddress address, JmDNSImpl dns, String jm
         if (index > 0) {
             aName = aName.substring(0, index);
         }
+        if (aName.length() > _labelLengthLimit) {
```

Comment:
Can we assume that if this situation happens, the discovered device does not correctly respect the specification?
If so, we should add at least some logging (warn or maybe better info level?).

Reply:
I don't think the device violates the spec. It's more the way that JmDNS converts an interface name to a single DNS label.
Concretely, I saw this problem in a situation where I had JmDNS running in an Azure pipeline job.
The Azure VM's network interface name is something like fv-az767-139.5kzmkpjakaaulcrpfkoga5aama.gx.internal.cloudapp.net. When the dots are replaced by dashed, it becomes a single DNS label which exceeds 63 characters. In the original interface name, each segment is less than 63 characters, and the whole name is less than 255 characters.